src/prompts/template.ts

```diff
@@ -57,22 +57,25 @@ export const templateMissingOptionPrompt = async (options, folderNameAnswers, de
       name: 'template',
       message: 'Please choose which node-mongo API boilerplate template to use',
       choices: templateCollection, //uses full names of templates in prompt for better user experience
-      default: defaultTemplate
+      default: defaultTemplate,
+      template: defaultTemplate
     });
   }
 
-  let templateAnswers;
+  let templateAnswers:any;
 
   if (options.skipPrompts) {
     options = skipPromptsModified(options, defaultFolderName, notAmongTemplateCollection, defaultTemplate);
-    templateQuestions.template = defaultTemplate;
+    if(templateQuestions[0] !== undefined){
+      templateQuestions[0].template = defaultTemplate;
+    }
```

Comment:
Is there a reason why you added this check here here for `templateQuestions[0]`? @tylerjusfly

Reply:
Not only the check, I see it's now totally different from what was there which is: `templateQuestions.template = defaultTemplate;`

Help change it back.

Reply:
at the beginning of the code there is an empty array
 const templateQuestions  = [];
 
then later in the code we pushed an object in the array like so

templateQuestions.push({
      type: 'list',
      name: 'template',
      message: 'Please choose which node-mongo API boilerplate template to use',
      choices: templateCollection, //uses full names of templates in prompt for better user experience
      default: defaultTemplate,
    });
    
    which means 
1. templateQuestions  is an array with an object
2. template does not exist in each objects in the array

then in the next block of code we are trying to access .template from templateQuestions which is an array and .template does not exist. 
  if (options.skipPrompts) {
    options = skipPromptsModified(options, defaultFolderName, notAmongTemplateCollection, defaultTemplate);
      templateQuestions.template = defaultTemplate;
    templateAnswers = templateQuestions;
  }
  so this code throws an error

Reply:
Ok, I will have time to check it better later today.

Reply:
@tylerjusfly this is probably an old code I forgot to remove. It literally does nothing significant to the code you can remove that line: 
`templateQuestions.template = defaultTemplate;`
> Remember to clean up all those spaces you added in the process of changing earlier.

Reply:
Solution for now @tylerjusfly, use this (for now):

1. Create a type
````
interface TemplateQuestions {
  type: string;
  name: string;
  message: string;
  choices: string[];
  default: string;
};
````

2. Use the type on the variable
````
 const templateQuestions: TemplateQuestions[] = [];
````

3. And then on that line you can use this:
````
(templateQuestions as any).template = defaultTemplate;
````
> **Note:** Remember to remove those extra spaces that you mistakenly added before. Let your code be clean and show the changes clearly.

#
**Explanation**
So the change you made before would have caused error if you did not specify the template in your command - e.g. using this in the terminal: ` node-mongo testFolder --yes`
That is because at the point where the template is not specified and the --yes flag is used to disable all  prompts, the default template is what will be bootstrapped by the CLI.

Interestingly, we have this funny shape of array below if you console.log `templateQuestions` at this point (I can see array having property and value of it's own for the first time in my life 😂):
````
[
  {
    type: 'list',
    name: 'template',
    message: 'Please choose which node-mongo API boilerplate template to use',
    choices: [
      'typescript API template (ts)',
      'es Module API template (esm)',
      'commonjs module API template (cjs)'
    ],
    default: 'typescript API template (ts)'
  },
  template: 'typescript API template (ts)'
]
````

Anyway, please use the solution I gave you before the explanation. We should deal with that when we start fixing the code proper I believe.